src/media/media.controller.ts

```diff
@@ -31,4 +36,12 @@ export class MediaController {
     // send ack response
     return { ack: true };
   }
+
+  @Get('/:id')
+  getMedia(@Param() params: GetMediaDTO, @Response() res) {
+    const fileName = this.mediaService.getMedia(params.id);
+    const x = path.join(__dirname, '../../../media/');
```

Comment:
- instead of joining paths like what in line 43 and having to write `../../../media/` which is prone to error. You could use path.resolve as it joins and resolves to the absolute path.
- Please check the suggested refactored code below: 
```typescript
getMedia(@Param() params: GetMediaDTO, @Response() res) {
    // get file name
    const fileName = this.mediaService.getMedia(params.id);

    // send file
    res.sendFile(path.resolve('media', fileName));
  }
```